qiskit_machine_learning/algorithms/classifiers/neural_network_classifier.py

```diff
@@ -119,4 +125,26 @@ def predict(self, X: np.ndarray) -> np.ndarray:  # pylint: disable=invalid-name
     def score(
         self, X: np.ndarray, y: np.ndarray, sample_weight: Optional[np.ndarray] = None
     ) -> float:
+        y = self._encode_categorical(y)
         return ClassifierMixin.score(self, X, y, sample_weight)
+
+    def _fit_and_encode_categorical(self, y: np.ndarray) -> np.ndarray:
+        """Fits label or one-hot encoder and converts categorical target data."""
+
+        if isinstance(y[0], str):
+            # string data is assumed to be categorical
+            y = y.reshape(-1, 1)
+            self._target_encoder.fit(y)
+            y = self._target_encoder.transform(y)
+
+        return y
+
+    def _encode_categorical(self, y: np.ndarray) -> np.ndarray:
+        """Converts categorical target data using label or one-hot encoding."""
+
+        if isinstance(y[0], str):
```

Comment:
This `if` statement really annoys me, any ideas how we can avoid it?

Reply:
I can't really see away around it - we need to check for categorical data somehow and looking for data of type `str` seems to be the cleanest way. I am always open to suggestions though.

What part bugs you?

Reply:
Well, if we know how labels were fitted and transformed in training then likely we know what to do here. That's why I'm thinking of this.

Reply:
I also don't have a neat solution, so let's keep it as it is now.
P.S. seems like the CI is stuck somewhere...